### ISSUE TITLE
Upgrade otter-grader to 4.2.0

### ISSUE THREAD

**Issue by sean-morris** [2i2c-org/infrastructure#1970]
### Context

The cloudbank pilot colleges are going to start using notebooks that are generated from or converted to the new otter-grader 4 version format. This requires the hub user image to also have otter-grader 4 installed for any college using the updated notebooks. 

We need to keep the current image with otter-grader 2.2.6 available for colleges that have not yet switched to new notebooks.

### Proposal

Create a new hub-user-image named hub-user-image-otter-4(I am open here to names) with otter-grader 4.2.0 installed.

### Updates and actions

- [x] create 2i2c-hubs-image-otter-4 repo cloned from the GH template repo, https://github.com/2i2c-org/hub-user-image-template or,  forked from, https://github.com/2i2c-org/2i2c-hubs-image. There is literally one change to the otter-grader version in the requirements.txt file so I forked https://github.com/2i2c-org/2i2c-hubs-image. Tell me I should use the template repo and just copy everything over from the 2i2c-hubs-image repo
- [x] In requirements.txt, change the otter-grader version to 4.2.0
- [ ] Obtain the quay.io secrets so GH Actions can push the new image to quay.io/2i2c(my quay username is: sean_smorris)
- [ ] I am a little unclear on how to assign a particular user image to a specific pilot hub without changing the helm charts in the infrastructure repo. I do get that I could use the configurator on the hub but I guess I am wondering what happens on a re-deploy of the hub? How does the change to the user image persist?

**Comment by yuvipanda** [comment]
(moved to the infrastructure image).

I would suggest that the new image be called 'cloudbank-hubs-image' or something like that? And could be kept up to date over time with the needs of the cloudbank hubs.

@sean-morris https://infrastructure.2i2c.org/en/latest/topic/infrastructure/config.html has more information on how the config made in the configurator persists! tldr is that it will persist across redeploys as well :)

**Comment by yuvipanda** [comment]
@sean-morris I created https://quay.io/repository/2i2c/cloudbank-image?tab=settings and gave you access too.

**Comment by ciara-morris** [comment]
@yuvipanda 
Re: github repo location.  Do you want the GH repo in the 2i2c organization? If so, I don't have permission to create a repository there. Otherwise, I could look into putting it with Data 8.

Re: image name -- "cloudbank-hubs-image" this is fine I guess the hub-user-image with otter-grader 2.2.6 is for general use. We will know the cloudbank image uses otter-grader 4.2.0

Great thanks

**Comment by sean-morris** [comment]
@yuvipanda
Re: github repo location. Do you want the GH repo in the 2i2c organization? If so, I don't have permission to create a repository there. Otherwise, I could look into putting it with Data 8.

Re: image name -- "cloudbank-hubs-image" this is fine I guess the hub-user-image with otter-grader 2.2.6 is for general use. We will know the cloudbank image uses otter-grader 4.2.0

Great thanks

**Comment by sean-morris** [comment]
My daughter was logged into GH on my computer! Hence the deleted comment I mistakenly posted from her!

**Comment by sean-morris** [comment]
@yuvipanda Circling back here. I see my account and the permissions for this image repository: https://quay.io/repository/2i2c/cloudbank-image?tab=settings

All good.

I have messed around and I don't think I can create a Robot account to be used to run the build action from GH. I tried a Robot account from my Quay account but had no ability to give the robot on my account permissions on the 2i2c/cloudbank-image quay repository.  I also tried my username and password as the Secret but that failed as well -- confused by that one -- I have to admit.

Here is the Invalid Username and Password log message from the GH Action:
https://github.com/2i2c-org/cloudbank-hubs-image/actions/runs/3783401190/jobs/6431911145

It looks like I need to create a Robot account on the 2i2c Quay account or get access to the name and password of a current Quay 2i2c Robot Account:
https://docs.quay.io/glossary/robot-accounts.html

I am not seeing that I have permission to do either.

I hope I have not missed some detail here.

Thanks
